xarray/backends/rasterio_.py

```diff
@@ -38,9 +38,7 @@ def shape(self):
         return self._shape
 
     def __getitem__(self, key):
-
-        # make our job a bit easier
-        key = indexing.canonicalize_indexer(key, self._ndims)
```

Comment:
I want to deprecate `indexing.canonicalize_indexer` because the same thing is done by `Variable._broadcast_indexes` method.

Currently, this is the only line that uses `indexing.canonicalize_indexer`.
Tests pass even without this line, but I don't fully understand what is happening here.

I would be very helpful if anyone comments on this.